### PR TITLE
Alex Novak - Homework 1 complete

### PR DESCRIPTION
**Pull request by PotatoSalads**
All tests passed and no errors with callbacks.

### CODE REVIEW

File: src/arrays.js [hunk 1]
```diff
@@ -9,36 +9,77 @@ const each = (elements, cb) => {
...
 };
 
 const reduce = (elements, cb, startingValue) => {
   // Combine all elements into a single value going from left to right.
   // Elements will be passed one by one into `cb` along with the `startingValue`.
   // `startingValue` should be the first argument passed to `cb` and the array element should be the second argument.
   // `startingValue` is the starting value.  If `startingValue` is undefined then make `elements[0]` the initial value.
+  if (startingValue === undefined) {
+    startingValue = elements[0];
+    for (let i = 1; i < elements.length; i++) {
```
Comment (SunJieMing) [inline src/arrays.js:34]: Right here you're repeating almost identical `for` loops.  Try to have one loop and then change some variables inside of the `if` block

File: src/arrays.js [hunk 2]
```diff
@@ -9,36 +9,77 @@ const each = (elements, cb) => {
...
 const find = (elements, cb) => {
   // Look through each value in `elements` and pass each element to `cb`.
   // If `cb` returns `true` then return that element.
   // Return `undefined` if no elements pass the truth test.
+  for (let i = 0; i < elements.length; i++) {
+    if (cb(elements[i])) {
+      return elements[i];
+    }
+  }
+  return 'undefined';
```
Comment (SunJieMing) [inline src/arrays.js:54]: Actually return the value `undefined` and not the string `'undefined'`.  Those are backticks around the word `undefined` in the description

File: src/callbacks.js [hunk 3]
```diff
@@ -30,51 +33,73 @@ firstItem(foods, (firstItem) => {
 });
 
 // Write a function called getLength that passes the length of the array into the callback
-// code here
+const getLength = (arr, cb) => {
+    cb(arr.length);
```
Comment (SunJieMing) [inline src/callbacks.js:37]: You are indented too far here

File: src/callbacks.js [hunk 4]
```diff
@@ -30,51 +33,73 @@ firstItem(foods, (firstItem) => {
...
+};
 
 last(foods, (lastItem) => {
   console.log(`The last item in the array is ${lastItem}.`);
 });
 
 // Write a function called sumNums that adds two numbers and passes the result to the callback
-// code here
+const sumNums = (num1, num2, cb) => {
+    cb(num1 + num2);
```
Comment (SunJieMing) [inline src/callbacks.js:55]: Same here

File: src/callbacks.js [hunk 5]
```diff
@@ -30,51 +33,73 @@ firstItem(foods, (firstItem) => {
...
+};
 
 sumNums(5, 10, (sum) => {
   console.log(`The sum is ${sum}.`);
 });
 
 // Write a function called multiplyNums that multiplies two numbers and passes the result to the callback
-// code here
+const multiplyNums = (num1, num2, cb) => {
+    cb(num1 * num2);
```
Comment (SunJieMing) [inline src/callbacks.js:64]: and here

File: src/callbacks.js [hunk 6]
```diff
@@ -30,51 +33,73 @@ firstItem(foods, (firstItem) => {
...
 
 contains(foods, 'ribeye', (result) => {
   console.log(result ? 'ribeye is in the array' : 'ribeye is not in the array');
 });
 
 // Write a function called removeDuplicates that removes all duplicate values from the given array.
 // Pass the array to the callback function.  Do not mutate the original array.
-// code here
+const removeDuplicates = (arr, cb) => {
+    const remArray = arr.sort();
```
Comment (SunJieMing) [inline src/callbacks.js:84]: A general principle is to keep the formatting as consistent as possible.  This makes your code look very clean

File: src/closure.js [hunk 7]
```diff
@@ -5,17 +5,31 @@ const counter = () => {
   // Example: const newCounter = counter();
   // newCounter(); // 1
   // newCounter(); // 2
+  let counts = 0;
+  return function newCounter() { return counts += 1; };
```
Comment (SunJieMing) [inline src/closure.js:9]: Generally when we are using the `function` keyword like this we don't condense it onto a single line

File: src/callbacks.js (hunk 3 above)
Comment (PotatoSalads) [reply src/callbacks.js:37]: thanks, i switched over to the webstorm IDE and didn't figure out the indenting/settings stuff yet. should be good for the next one

File: src/closure.js [hunk 8]
```diff
@@ -5,17 +5,31 @@ const counter = () => {
...
 
 const limitFunctionCallCount = (cb, n) => {
   // Should return a function that invokes `cb`.
   // The returned function should only allow `cb` to be invoked `n` times.
+  let num = 0;
+  return (...args) => {
+    if (num < n) {
+      num++;
+      return cb(...args);
+    } return null;
```
Comment (SunJieMing) [inline src/closure.js:31]: This `return` should be on the next line

**Pull request closed without merging**